src/generators/python/index.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { IDocumentWidget } from '@jupyterlab/docregistry';
+import { FileEditor, IEditorTracker } from '@jupyterlab/fileeditor';
+import { TableOfContentsRegistry as Registry } from '../../registry';
+import { IHeading } from '../../utils/headings';
+import { render } from './render';
+
+/**
+ * Generates a table of contents.
+ *
+ * @private
+ * @param editor - editor widget
+ * @returns a list of headings
+ */
+function generate(editor: IDocumentWidget<FileEditor>): IHeading[] {
+  // Split the text into lines:
+  let lines = editor.content.model.value.text.split('\n') as Array<any>;
+
+  // Iterate over the lines to get the heading level and text for each line:
+  let headings: IHeading[] = [];
+  let processingImports = false;
+  for (let i = 0; i < lines.length; i++) {
+    let line = lines[i].trimLeft();
+    if (line.indexOf('def ') === 0) {
+      processingImports = false;
+      headings.push({
+        text: line.slice(0, -1),
```

Comment:
Above you are doing a trimLeft, which means that if there are trailing spaces on the right then `:` will not be handled properly.